### PR TITLE
Deprecate Closure taking methods

### PR DESCRIPTION
**Pull request by wolfs**
Any of the checked boxes below indicate that I took action:

- [ ] Reviewed the [Contribution Guidelines](https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md#contribution-workflow).
- [ ] Signed the [Gradle CLA](http://gradle.org/contributor-license-agreement/).
- [ ] Ensured that basic checks pass: `./gradlew quickCheck`

For all non-trivial changes that modify the behavior or public API:

- [ ] Before submitting a pull request, I started a discussion on the
[Gradle developer list](https://groups.google.com/forum/#!forum/gradle-dev)
or can reference a [JIRA issue](https://issues.gradle.org/secure/Dashboard.jspa).
- [ ] I wrote a design document. A design document can be
brief but explains the use case or problem you are trying to solve,
touches on the planned implementation approach as well as the test cases
that verify the behavior. Optimally, design documents should be submitted
as a separate pull request. [Samples](https://github.com/gradle/gradle/tree/master/design-docs)
can be found in the Gradle GitHub repository. Please let us know if you need help with
creating the design document. We are happy to help!
- [ ] The pull request contains an appropriate level of unit and integration
test coverage to verify the behavior. Before submitting the pull request
I ran a build on your local machine via the command
`./gradlew quickCheck <impacted-subproject>:check`.
- [ ] The pull request updates the Gradle documentation like user guide,
DSL reference and Javadocs where applicable.

### CODE REVIEW

File: subprojects/core/src/main/java/org/gradle/api/tasks/TaskOutputs.java [hunk 1]
```diff
@@ -38,8 +38,12 @@
      * <p>You can add multiple such predicates. The task outputs are considered out-of-date when any predicate returns
      * false.<p>
      *
+     * @deprecated Use {@link #upToDateWhen(Spec)}
+
```
Comment (bmuschko) [inline subprojects/core/src/main/java/org/gradle/api/tasks/TaskOutputs.java]: Extra whitespace.

File: subprojects/core/src/main/java/org/gradle/api/tasks/TaskOutputs.java [hunk 2]
```diff
@@ -38,8 +38,12 @@
...
      * @param upToDateClosure The closure to use to determine whether the task outputs are up-to-date.
+     *
```
Comment (bmuschko) [inline subprojects/core/src/main/java/org/gradle/api/tasks/TaskOutputs.java]: Unnecessary new line.

**Comment by bmuschko** [comment]
- We should probably document SAM-type coercion in the [Groovy basics section](https://docs.gradle.org/current/userguide/writing_build_scripts.html#groovy-dsl-basics).
- What about other public interfaces that take a `Closure` parameter e.g. `SourceSet`, `TaskContainer`?

**Comment by wolfs** [comment]
Deprecation is not the right solution here as it will prompt build file authors that they are using deprecated methods without a viable alternative.

**Pull request closed without merging**